pkg/ddc/jindofsx/transform.go

```diff
@@ -819,45 +820,87 @@ func (e *JindoFSxEngine) transformFuseArg(runtime *datav1alpha1.JindoRuntime, da
 	return fuseArgs
 }
 
-func (e *JindoFSxEngine) getSmartDataConfigs() (image, tag, dnsServer string) {
+func (e *JindoFSxEngine) getSmartDataConfigs(runtime *datav1alpha1.JindoRuntime) (image, tag, imagePullPolicy, dnsServer string) {
 	var (
-		defaultImage     = "registry.cn-shanghai.aliyuncs.com/jindofs/smartdata"
-		defaultTag       = "4.5.2"
-		defaultDnsServer = "1.1.1.1"
+		defaultImage           = "registry.cn-shanghai.aliyuncs.com/jindofs/smartdata"
+		defaultTag             = "4.5.2"
+		defaultImagePullPolicy = "Always"
+		defaultDnsServer       = "1.1.1.1"
 	)
 
-	image = docker.GetImageRepoFromEnv(common.JindoSmartDataImageEnv)
-	tag = docker.GetImageTagFromEnv(common.JindoSmartDataImageEnv)
-	dnsServer = os.Getenv(common.JindoDnsServer)
-	if len(image) == 0 {
-		image = defaultImage
+	// Apply defaults
+	image = defaultImage
+	tag = defaultTag
+	imagePullPolicy = defaultImagePullPolicy
+	dnsServer = defaultDnsServer
+
+	// Override with global-scoped configs
+	globalImage := docker.GetImageRepoFromEnv(common.JindoSmartDataImageEnv)
+	globalTag := docker.GetImageTagFromEnv(common.JindoSmartDataImageEnv)
+	globalDnsServer := os.Getenv(common.JindoDnsServer)
+	if len(globalImage) > 0 {
+		image = globalImage
+	}
+	if len(globalTag) > 0 {
+		tag = globalTag
+	}
+	if len(globalDnsServer) > 0 {
+		dnsServer = globalDnsServer
 	}
-	if len(tag) == 0 {
-		tag = defaultTag
+
+	// Override with runtime-scoped configs
+	if len(runtime.Spec.JindoVersion.Image) > 0 {
```

Comment:
Use switch case instead of if else to improve the readability of the code.
```golang
switch {
case len(runtime.Spec.JindoVersion.Image) > 0:
	image = runtime.Spec.JindoVersion.Image
case len(runtime.Spec.JindoVersion.ImageTag) > 0:
	tag = runtime.Spec.JindoVersion.ImageTag
case len(runtime.Spec.JindoVersion.ImagePullPolicy) > 0:
	imagePullPolicy = runtime.Spec.JindoVersion.ImagePullPolicy
}
```